sqflite/lib/src/connection.dart

```diff
@@ -0,0 +1,194 @@
+    return connection.rawQuery(stStr);
+  }
+
+  /// Executes the insert statement and returns the primary key of
+  /// inserted row
+  @override
+  Future<T> insert<T>(Insert st) {
+    String strSt = composeInsert(st);
+    if (logger != null) logger(strSt);
+    return connection.rawInsert(strSt) as Future<T>;
+  }
+
+  /// Executes the insert statement for many element
+  @override
+  Future<void> insertMany<T>(InsertMany st) async {
+    String strSt = composeInsertMany(st);
+    return exec(strSt);
+  }
+
+  /// Executes the insert or update statement and returns the primary key of
+  /// inserted row
+  @override
+  Future<T> upsert<T>(Upsert st) {
+    String strSt = composeUpsert(st);
+    if (logger != null) logger(strSt);
+    return connection.rawInsert(strSt) as Future<T>;
+  }
+
+  /// Executes bulk insert or update statement
+  @override
+  Future<void> upsertMany<T>(UpsertMany st) {
+    List<String> strSt = composeUpsertMany(st);
+    if (logger != null) logger(strSt.toString());
+    final batch = connection.batch();
+    for (var query in strSt) {
+      batch.execute(query);
+    }
+    return batch.commit(noResult: true);
+  }
+
+  /// Updates a record in the table
+  @override
+  Future<int> update(Update st) {
+    String strSt = composeUpdate(st);
+    if (logger != null) logger(strSt);
+    return connection.rawUpdate(strSt);
+  }
+
+  /// Deletes a record from the table
+  @override
+  Future<int> remove(Remove st) {
+    String sql = composeRemove(st);
+    if (logger != null) logger(sql);
+    return connection.rawDelete(sql);
+  }
+
+  @override
+  Future<void> alter(Alter statement) async {
+    await exec(composeAlter(statement));
+  }
+
+  /// Creates the table
+  @override
+  Future<void> createTable(Create statement) async {
+    await exec(composeCreate(statement));
+  }
+
+  /// Create the database
+  @override
+  Future<void> createDatabase(CreateDb st) async {
+    await exec(composeCreateDb(st));
+  }
+
+  /// Drops tables from database
+  @override
+  Future<void> dropTable(Drop st) async {
+    String stStr = composeDrop(st);
+    await exec(stStr);
+  }
+
+  /// Drops tables from database
+  @override
+  Future<void> dropDb(DropDb st) async {
+    String strSt = composeDropDb(st);
+    await exec(strSt);
+  }
+
+  /// Parses values coming from database into Dart values
+  @override
+  T parseValue<T>(dynamic v) {
+    if (T == String) {
+      return v;
+    } else if (T == int) {
+      return v?.toInt();
+    } else if (T == double) {
+      return v?.toDouble();
+    } else if (T == num) {
+      return v;
+    } else if (T == DateTime) {
+      if (v == null) return null;
+      if (v is String) return DateTime.parse(v) as T;
+      if (v == int) return DateTime.fromMillisecondsSinceEpoch(v * 1000) as T;
+      return null;
+    } else if (T == bool) {
+      if (v == null) return null;
+      return (v == 0 ? false : true) as T;
+    } else {
+      return v as T;
```

Comment:
any reason for doing this and not throwing an error like previously ?

Reply:
I had one really ignoring issue where I defined a field as an ignored list.
I also saw that postgres was doing only casting so I assumed I can do the same thing if type is not one of the known types